### PR TITLE
Allow same section class for multiple section types (fix #10764)

### LINKED ISSUE #10764: Custom setting section not visible if used for admin and personal setting in NC14
In NC 13 everything works as expected, but in NC 14 my admin section is missing (the personal section is there). If I remove the `personal-section` element, the admin section is there. @juliushaertl thinks this is maybe a bug.

Here's an example of my `info.xml`:
https://github.com/nextcloud/jsxc.nextcloud/blob/9f342ec4455ad67f9960a152f28894dc22bd3490/appinfo/info.xml#L45-L50

ref jsxc/jsxc#718

### Steps to reproduce
1. Use the same section in your `info.xml` for `admin-section` and `personal-section`
2. Go to settings in your NC

### Expected behaviour
There is a custom section in personal and admin section.

### Actual behaviour
Only the personal section is visible.

### Server configuration

**Operating system**: ubuntu 18.04

**Web server:** apache

**Database:** sqlite

**PHP version:** 7.2

**Nextcloud version:** 14 beta.4

**Updated from an older Nextcloud/ownCloud or fresh install:** fresh

**Where did you install Nextcloud from:** git

### Client configuration
**Browser:** Vivaldi

**Operating system:** ubuntu 18.04

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

Nothing relevant.
</details>

#### Browser log
<details>
<summary>Browser log</summary>

Nothing.
</details>

### PR DESCRIPTION
**Pull request by sualko**
This fixes a regression introduced in the upcoming NC 14.

### CODE REVIEW

File: lib/private/Settings/Manager.php [hunk 1]
```diff
@@ -145,7 +149,11 @@ protected function getSections(string $type): array {
 			$this->sections[$type] = [];
 		}
 
-		foreach ($this->sectionClasses as $class => $sectionType) {
+		if (!isset($this->sectionClasses[$type])) {
+			return $this->sections[$type];
+		}
+
+		foreach ($this->sectionClasses[$type] as $index => $class) {
```
Comment (nickvergessen) [inline lib/private/Settings/Manager.php]: `foreach ($this->sectionClasses[$type] as $class) {` works in php

Comment (sualko) [reply lib/private/Settings/Manager.php]: yes, but I need the index a couple of lines below

Comment (nickvergessen) [reply lib/private/Settings/Manager.php]: Ah in the unset.

**Review by blizzz — APPROVED** [review]
looks good and works

**Review by MorrisJobke — APPROVED** [review]

**Pull request merged**